### DISCUSSION TITLE
咨询如何将录音的音频文件上传至云端进行语音识别，或者本地是否有识别库？

### DISCUSSION THREAD

**Discussion by xkkjiayou** [zepp-health/zeppos-forum#299] [Q&A]
咨询如何将录音的音频文件上传至云端进行语音识别，或者本地是否有识别库？
我尝试将录音文件OPUS转为base64，发现好像zeppos不支持。
尝试直接通过fetch API将文件包裹在formData中进行上传，也同样不支持。
请问该如何处理，这种情况
我正在开发一个 语音对话的应用。

来自中科院医学团队，谢谢
我的邮箱 xkk9866@yeah.net

**Comment by xkkjiayou** [comment]
zeppOS 3， balance系列手表

**Reply by htoooth** [reply]
‌‌I have replied to you in a private channel.
@xkkjiayou

**Reply by buertang** [reply]
你好，请问是什么方式呢？

**Comment by xkkjiayou** [comment]
感谢您的来信，此时我不在办公室，稍后会及时阅读。在此期间我不方便接收邮件。如果您想及时获得通讯，请拨打我的手机联系我，谢谢您。

**Reply by buertang** [reply]
你好，是怎么解决的呢？我也需要将录音的音频文件上传。

**Comment by beegweeg** [comment]
I do things like this. First, I record the file using the recorder from this documentation: [Zepp Recorder API](https://docs.zepp.com/docs/reference/device-app-api/newAPI/media/). Then I send it using `Fetch`.

**Page:**
```javascript
getFileBuffer(filePath) {
  const res = readFileSync({
    path: filePath,
  });
  return Buffer.from(res);
}
```

**App Side:**
```javascript
const fileBuffer = req.fileBuffer;
const headers = { "Content-Type": "application/json" };
const response = await fetch(url, {
  method: 'POST',
  headers,
  body: fileBuffer
});
```

On the backend:
```javascript
const path = 'test.opus';
const buffer = Buffer.from(req.body);
await fs.writeFile(path, buffer);
convertToWav(path, outputPath);
```

The `convertToWav` function is taken from the docs: [Zepp FAQ: Convert OPUS to MP3](https://docs.zepp.com/docs/guides/faq/opus-to-mp3/?_highlight=convert/docs/guides/faq/opus-to-mp3/#steps).
